### PR TITLE
chore: expected<Something, Error> should never allow construction from Error without unexpected()

### PR DESCRIPTION
**Pull request by lukfi**
# External Contributor Checklist

🚨 Please review the [guidelines for contributing](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md) to this repository.

- [ ] Make sure that all commits follow the conventional commit [specification](https://www.github.com/mendersoftware/mendertesting/commitlint/grammar.md) for the Mender project.

The majority of our contributions are fixes, which means your commit should have
the form below:

```
fix: <SHORT DESCRIPTION OF FIX>

<OPTIONAL LONGER DESCRIPTION>

Changelog: <USER-FRIENDLY-CHANGE-DESCRIPTION> or <None>
Ticket: <TICKET NUMBER> or <None>
```

- [ ] Make sure that all commits are signed with [`git --signoff`](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work). Also note that the signoff author must match the author of the commit.

### Description

Please describe your pull request.

Thank you!

### CODE REVIEW

**Review by oleorhagen — APPROVED** [review]
Looks good imo.

Clean up the commits a little maybe :)

File: common/error.hpp [hunk 1]
```diff
@@ -53,8 +53,8 @@ class Error {
 		return this->message == other.message && this->code == other.code;
 	}
 
-	operator bool() const {
-		return static_cast<bool>(this->code);
+	bool operator!=(const Error &other) const {
+		return this->message != other.message || this->code != other.code;
```
Comment (vpodzime) [inline common/error.hpp:57]: How does this work in comparisons with `error::NoError` when the `code` is `error_condition` and thus has both a category and an integer code?

Comment (lukfi) [reply common/error.hpp:57]: Yes, we return error::NoError in all the places.
I've cleaned up the commits.

**Comment by kacf** [comment]
@lukfi: I suspect that the merging of [this PR](https://github.com/mendersoftware/mender/pull/1189) might create some new cases which need to be converted. Sorry. 🙂

**Pull request merged**